packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -192,16 +190,20 @@ export const MenuItem = React.forwardRef<HTMLLIElement, MenuItemProps>(
         <Popover
           placement='right-start'
           visible={isSubmenuVisible}
-          appendTo='parent'
+          appendTo={(item) => {
+            // we want the div wrapping the parent menu because the menu has overflow,
+            // which can cause the submenu to be cut off
+            return item.parentElement?.parentElement as HTMLElement;
+          }}
           delay={100}
           content={
             <div
               onMouseLeave={() => setIsSubmenuVisible(false)}
-              onBlur={(e) => {
-                !!(e.relatedTarget instanceof Node) &&
-                  !subMenuRef.current?.contains(e.relatedTarget as Node) &&
-                  !subMenuRef.current?.isEqualNode(e.relatedTarget as Node) &&
+              onKeyDown={(e) => {
+                if (!e.altKey && e.key === 'ArrowLeft') {
```

Comment:
To confirm, is this so that the browser's "go back" keyboard shortcut will take precedence? If so, what will happen if the user uses a browser that has some other shortcut? Or maybe the user has overridden the shortcut somehow to something else?

Since the <kbd>Alt</kbd> equivalent on the Mac is <kbd>Option</kbd>, but the "go back" shortcut is <kbd>Command</kbd>+<kbd>←</kbd>, is the browser "go back" keybaord shortcut given preference on Mac too?

Reply:
i'm assuming you're asking about `e.altKey`, so . i've kept that part the same as before, so it doesn't work for Mac currently.

if we improve this someday, we should probably do it for all components